### PR TITLE
Source chain injection

### PR DESCRIPTION
**Pull request by freesig**
### Summary

- Adds the ability to manually insert elements into a source chain using the `AdminRequest::AddElements` command. Please check the docs for more details / warnings on proper usage.

### TODO:
- [x] CHANGELOG(s) updated with appropriate info
- [ ] Just before pressing the merge button, ensure new entries to CHANGELOG(s) are still under the _UNRELEASED_ heading

### CODE REVIEW

File: crates/holochain/src/conductor/api/api_external/admin_interface.rs [hunk 1]
```diff
@@ -331,6 +331,18 @@ impl AdminInterfaceApi for RealAdminInterfaceApi {
                 self.handle_admin_request_inner(DisableApp { installed_app_id })
                     .await
             }
+            AddElements {
+                cell_id,
+                elements,
+                truncate,
+                validate,
```
Comment (neonphog) [inline crates/holochain/src/conductor/api/api_external/admin_interface.rs:338]:
```suggestion
                truncate,
                validate,
                elements,
```

It's in this order everywhere else... : )

**Review by neonphog — APPROVED** [review]
:+1:

File: crates/holochain_conductor_api/src/admin_interface.rs [hunk 2]
```diff
@@ -298,6 +298,48 @@ pub enum AdminRequest {
         /// Optionally choose a specific agent info
         cell_id: Option<CellId>,
     },
+
+    /// Insert [`Element`]s into the source chain of the [`CellId`].
+    ///
+    /// All elements must be authored and signed by the same agent.
+    /// The [`Dna`] (but not necessarily the cell) must already exist
```
Comment (maackle) [inline crates/holochain_conductor_api/src/admin_interface.rs:305]:
```suggestion
    /// The [`DnaFile`] (but not necessarily the cell) must already be installed
```

**Review by maackle — APPROVED** [review]
Nice work!

File: crates/holochain/src/conductor/handle.rs [hunk 3]
```diff
@@ -1224,6 +1238,167 @@ impl<DS: DnaStore + 'static> ConductorHandleT for ConductorHandleImpl<DS> {
...
+                        #[allow(clippy::needless_collect)]
+                        let basis = ops
+                            .iter()
+                            .map(|op| op.dht_basis().clone())
+                            .collect::<Vec<_>>();
+                        ops_to_integrate.extend(
+                            source_chain::put_raw(txn, shh, ops, entry)?
+                                .into_iter()
+                                .zip(basis.into_iter()),
+                        );
```
[the remark this answers is not in the document]
Comment (freesig) [reply crates/holochain/src/conductor/handle.rs:1378]: No because ops is moved on line 1372 and also borrowed on line 1369.

**Pull request merged**